### DISCUSSION TITLE
How to implement UI refresh upon window size change detection?

### DISCUSSION THREAD

**Discussion by zhangguolin333** [zauberzeug/nicegui#3578] [Q&A] [answered]
```py
@ui.page('/index_page')
def index_page():
    create_dialogs()
    create_header()
    create_main_layout()
    create_footer()

app.native.window_args['resizable'] = True
app.native.window_args['maximized'] = True
app.native.start_args['debug'] = False
app.native.settings['ALLOW_DOWNLOADS'] = True

index_page()

ui.run(title='DataView V 0.0.1', reload=False, port=native.find_open_port(), native=True, window_size=(1250, 750))
```

![image](https://github.com/user-attachments/assets/92f0583a-128f-41f6-839c-9b221932dacc)
after maximized:
![image](https://github.com/user-attachments/assets/48e88262-3cbb-44fa-ac92-5e45b2916c2d)

**Comment by falkoschindler** [comment]
@zhangguolin333 What exactly do you want to happen?
And can you, please, share a minimum executable example so we can reproduce the behavior? Thanks!
[Why is it important to provide a minimal reproducible example?](https://github.com/zauberzeug/nicegui/wiki/FAQs#why-is-it-important-to-provide-a-minimal-reproducible-example)

**Reply by python-and-novella** [reply]
I guess he want to handle this in nicegui:
https://pywebview.flowrl.com/guide/api.html#window-events-resized

**Reply by python-and-novella** [reply]
It seems that we need to wait native window created, then  run it:
```python3
app.native.main_window.events.resized += lambda :ui.notify('resized!')
```

**Comment by python-and-novella** [comment, marked as answer]
```python3
@ui.page('/')
async def init():
    await ui.context.client.connected()
    size = await app.native.main_window.get_size()
    label = ui.label(f'width is {size[0]}, height is {size[1]}.')
    ui.on('resize',lambda e: label.set_text(f'width is {e.args["width"]}, height is {e.args["height"]}.'))
```
